src/MagnumPlugins/TinyGltfImporter/TinyGltfImporter.cpp

```diff
@@ -300,33 +300,77 @@ Containers::Optional<MeshData3D> TinyGltfImporter::doMesh3D(const UnsignedInt id
     }
     /* LCOV_EXCL_STOP */
 
-    /* Vertices */
-    const tinygltf::Accessor& posAccessor = _d->model.accessors[primitive.attributes.find("POSITION")->second];
-    const tinygltf::BufferView& posBufferview = _d->model.bufferViews[posAccessor.bufferView];
-    const tinygltf::Buffer& posBuffer = _d->model.buffers[posBufferview.buffer];
-
-    if(posAccessor.type != TINYGLTF_TYPE_VEC3) {
-        Error() << "Trade::TinyGltfImporter::mesh3D(): expected type of vertex positions is VEC3";
-        return Containers::NullOpt;
-    }
-
     std::vector<Vector3> positions;
-    size_t numPositions = posBufferview.byteLength/sizeof(Vector3);
-    std::copy_n(reinterpret_cast<const Vector3*>(posBuffer.data.data() + posBufferview.byteOffset), numPositions, std::back_inserter(positions));
+    std::vector<Vector3> normals;
+    std::vector<std::vector<Vector2>> textureLayers;
+    std::vector<std::vector<Color4>> colorLayers;
 
-    const tinygltf::Accessor& normalAccessor = _d->model.accessors[primitive.attributes.find("NORMAL")->second];
-    const tinygltf::BufferView& normalBufferview = _d->model.bufferViews[normalAccessor.bufferView];
-    const tinygltf::Buffer& normalBuffer = _d->model.buffers[normalBufferview.buffer];
+    for (auto& attribute : primitive.attributes) {
+        const tinygltf::Accessor& accessor = _d->model.accessors[attribute.second];
+        const tinygltf::BufferView& bufferView = _d->model.bufferViews[accessor.bufferView];
+        const tinygltf::Buffer& buffer = _d->model.buffers[bufferView.buffer];
 
-    if(normalAccessor.type != TINYGLTF_TYPE_VEC3) {
-        Error() << "Trade::TinyGltfImporter::mesh3D(): expected type of normal is VEC3";
-        return Containers::NullOpt;
+        if (attribute.first.compare("POSITION") == 0) {
+            if(accessor.type != TINYGLTF_TYPE_VEC3) {
+                Error() << "Trade::TinyGltfImporter::mesh3D(): expected type of" << attribute.first << "is VEC3";
+                return Containers::NullOpt;
+            }
+
+            const size_t numPositions = bufferView.byteLength/sizeof(Vector3);
+            positions.reserve(numPositions);
+            std::copy_n(reinterpret_cast<const Vector3*>(buffer.data.data() + bufferView.byteOffset), numPositions, std::back_inserter(positions));
+
+        } else if (attribute.first.compare("NORMAL") == 0) {
+            if(accessor.type != TINYGLTF_TYPE_VEC3) {
+                Error() << "Trade::TinyGltfImporter::mesh3D(): expected type of" << attribute.first << "is VEC3";
+                return Containers::NullOpt;
+            }
+
+            const size_t numNormals = bufferView.byteLength/sizeof(Vector3);
+            normals.reserve(numNormals);
+            std::copy_n(reinterpret_cast<const Vector3*>(buffer.data.data() + bufferView.byteOffset), numNormals, std::back_inserter(normals));
+
+        /* Texture coordinate attribute ends with _0, _1 ... */
+        } else if (attribute.first.compare(0, 8, "TEXCOORD") == 0) {
```

Comment:
You could use `Utility::String::beginsWith()` here ;) Similarly for colors below.